kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/AbstractWatchManager.java

```diff
@@ -250,33 +248,9 @@ private WatchEvent contextAwareWatchEventDeserializer(String messageSource)
     }
   }
 
-  protected WatchEvent readWatchEvent(String messageSource) throws JsonProcessingException {
-    WatchEvent event = contextAwareWatchEventDeserializer(messageSource);
-    KubernetesResource object = null;
-    if (event != null) {
-      object = event.getObject();
-    }
-    // when watching API Groups we don't get a WatchEvent resource
-    // so the object will be null
-    // so lets try parse the message as a KubernetesResource
-    // as it will probably be a list of resources like a BuildList
-    if (object == null) {
-      object = Serialization.unmarshal(messageSource, KubernetesResource.class);
-      if (event == null) {
-        event = new WatchEvent(object, "MODIFIED");
-      } else {
-        event.setObject(object);
-      }
-    }
```

Comment:
What happens if you watch API Groups in a real cluster?

From yesterday discussion, the assumption was that receiving a list as an event was not possible, but the former comment here says otherwise.

Reply:
> What happens if you watch API Groups in a real cluster?

I couldn't quite determine what that comment meant.  You cannot watch APIGroup directly, and we have no example, including Builds, where a watch returns a list rather than a WatchEvent.  My only guess is that this was from early behavior in OpenShift and the comment could be referencing not using oapi for the watch.

Reply:
OK, I recall now you did provide that explanation.